packages/loopring_v3/circuit/Circuits/AmmUpdateCircuit.h

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+#ifndef _AMMUPDATECIRCUIT_H_
+#define _AMMUPDATECIRCUIT_H_
+
+#include "Circuit.h"
+#include "../Utils/Constants.h"
+#include "../Utils/Data.h"
+
+#include "ethsnarks.hpp"
+#include "utils.hpp"
+
+using namespace ethsnarks;
+
+namespace Loopring
+{
+
```

Comment:
Add a comment: Brings a token's L2 balance to Layer1.

Question1: is this tx only available to AMM account? 

Question2: besides the nonce, what data will also be changed after such a tx?

Reply:
> is this tx only available to AMM account?

This transaction is available to all account owners at any time. There isn't really a thing like an AMM account. AMM orders between 2 tokens are enabled on an account when those 2 tokens have a non-zero weight, that's the only requirement.

>  besides the nonce, what data will also be changed after such a tx?

Only the expected:
- AMM fee bips stored in the account
- AMM weight of the specified token

This transaction simply sets what's defined in the transaction itself in the Merkle tree, no other checks are done. Data is just copied to the Merkle tree.